### PR TITLE
Refactor header layout & fix header wording

### PR DESCRIPTION
**Pull request by user512**
Fix top-menu text according to mockup
* Ref: https://soloish.slack.com/archives/C013U0E797Y/p1598652883006900
* Ref: https://xd.adobe.com/view/fd425dbe-5384-44c9-997a-eeee6e886a86-a811/grid

Avoid duplication of similar layout and favor `content_for` technique

### CODE REVIEW

**Comment by zspencer** [comment]
One thing we may want to do to allow our CSS to target particular controllers or what not, is to include the controller name and action name as classes and attach them to the body in a block-element-modifier style. For example, when `GET '/workspaces/1/rooms/2'` 
```

<body class="<%= page_bem_class %>"
...
</body>
```
# Because the name of the controller is `Room` and the action is `show` it would be
<body class="room --show">

</body>
```

Then, we can write CSS to hide or show elements like so:

```
.room .top-menu {
  display: none;
}
```

Without having to update additional layouts or pass data back and forth between views and layouts.

File: convene-web/app/views/rooms/show.html.erb [hunk 1]
```diff
@@ -1,3 +1,11 @@
+<% content_for :stylesheets do %>
+  .top-menu { display: none }
+<% end %>
+
+<% content_for :head do %>
+  <title><%= "#{current_workspace.name} - #{current_room.name}" %></title>
```
Comment (zspencer) [inline convene-web/app/views/rooms/show.html.erb:6]: We may want an application helper for `page_title`, which we can either override on a per-controller basis or let stand.

**Review by zspencer — CHANGES_REQUESTED** [review]
I think this is a good stab at it, but I think we could tighten this up so there are fewer seams.

**Comment by user512** [comment]
Tried using this approach (https://github.com/zinc-collective/convene/pull/125#issuecomment-694488080) but then there isn't an out of the box way to expose template name (can easily expose controller_name & action_name) and our logic for menu placement is tied to the template.

Thought about exposing an instance variable by monkey patching ActionView but then it seems overdone and feel good about using `content_for` in `show.html.erb` to add an `app_no-top-menu` class because it's explicit and follow BEM naming methodology.

**Review by zspencer — APPROVED** [review]
This is much better! The things I like about it is we are adding fewer seams where things could go sideways.

A small completely non-blocking critique: I'm not entirely sold on the names of the `top-menu` files. I think `_header.scss` and `_header.html.erb` both are the logical "next step" in the "semantic tree" of the application layout. 

For example, most applications have the following  pieces:

```
- header
- navigation
- body
- footer
```

By keeping things named in a way that match these common usages, it makes it reduces the amount of guess-work a developer needs to engage in when figuring out where to make changes.

File: convene-web/app/javascript/src/page_bem_class.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.app {
+  &--no-top-menu {
+    .top-menu {
+      display: none;
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (user512) [reply convene-web/app/javascript/src/page_bem_class.scss:7]: I think this is the SCSS way of writing `.app.--no-top-menu`.
I learned this from https://css-tricks.com/the-sass-ampersand/#adding-another-class.

File: convene-web/app/views/rooms/show.html.erb [hunk 3]
```diff
@@ -1,3 +1,7 @@
+<% content_for :page_bem_class do %>
+  app--no-top-menu
+<% end %>
```
[the remark this answers is not in the document]
Comment (user512) [reply convene-web/app/views/rooms/show.html.erb:3]: I wasn't sure why it's throwing syntax error when trying the one-liner, I supposed it's related to how erb works but not sure exactly why it's failing, keeping a multiple line block for now.

File: convene-web/app/javascript/src/page_bem_class.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (user512) [reply convene-web/app/javascript/src/page_bem_class.scss:7]: But please let me know if I misunderstood.

**Pull request merged**